integration_tests/scheduled_cork_test.go

```diff
@@ -0,0 +1,205 @@
+				s.T().Logf("managed addresses: %v", res.CellarIds)
+				if common.HexToAddress(id) == counterContract {
+					found = true
+					break
+				}
+			}
+
+			return found
+		}, 10*time.Second, 2*time.Second, "did not find address in managed cellars")
+
+		s.T().Log("schedule a cork for the future")
+		node, err := clientCtx.GetNode()
+		s.Require().NoError(err)
+		status, err := node.Status(context.Background())
+		s.Require().NoError(err)
+		currentBlockHeight := status.SyncInfo.LatestBlockHeight
+		targetBlockHeight := currentBlockHeight + 15
+
+		s.T().Log("scheduling cork calls")
+		for i, orch := range s.chain.orchestrators {
+			i := i
+			orch := orch
+			s.Require().Eventuallyf(func() bool {
+				clientCtx, err := s.chain.clientContext("tcp://localhost:26657", orch.keyring, "orch", orch.keyInfo.GetAddress())
+				s.Require().NoError(err)
+
+				corkMsg, err := types.NewMsgScheduleCorkRequest(
+					ABIEncodedInc(),
+					counterContract,
+					uint64(targetBlockHeight),
+					orch.keyInfo.GetAddress())
+				s.Require().NoError(err, "unable to create cork schedule msg")
+
+				response, err := s.chain.sendMsgs(*clientCtx, corkMsg)
+				if err != nil {
+					s.T().Logf("error: %s", err)
+					return false
+				}
+				if response.Code != 0 {
+					if response.Code != 32 {
+						s.T().Log(response)
+					}
+					return false
+				}
+
+				s.T().Logf("cork msg for %d node sent successfully", i)
+				return true
+			}, 10*time.Second, 500*time.Millisecond, "unable to deploy cork schedule msg for node %d", i)
+		}
+
+		s.T().Logf("wait for the block to pass, monitoring it as it goes")
+		s.Require().Eventuallyf(func() bool {
+			kb, err := val.keyring()
+			s.Require().NoError(err)
+			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+			s.Require().NoError(err)
+
+			queryClient := types.NewQueryClient(clientCtx)
+			res, err := queryClient.QueryScheduledCorks(context.Background(), &types.QueryScheduledCorksRequest{})
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+
+			node, err := clientCtx.GetNode()
+			s.Require().NoError(err)
+			status, err := node.Status(context.Background())
+			s.Require().NoError(err)
+			blockHeight := status.SyncInfo.LatestBlockHeight
+
+			count, err = s.getCurrentCount()
+			if err != nil {
+				return false
+			}
+
+			if blockHeight < (targetBlockHeight - 2) {
+				// verify that tbe scheduled cork has not yet been consumed, and that the counter has not been incremented
+				s.Require().Len(res.Corks, len(s.chain.validators))
+				s.Require().Equal(counterContract, common.HexToAddress(res.Corks[0].Cork.TargetContractAddress))
+				s.Require().Equal(int64(0), count.Int64())
+			} else if blockHeight > (targetBlockHeight + 1) {
+				// verify that block height has been passed, cork consumed and counter is incremented
+				s.Require().Len(res.Corks, 0)
+				s.Require().Equal(int64(1), count.Int64())
```

Comment:
The counter check bit needs to be moved to a loop after this. The condition we're looking for here is the presence of a ContractCallTx on the chain (there is a query endpoint for this), it will take some time for the call to propagate over the bridge and update the contract.

Reply:
Moved to a separate loop, replaced it with a check for contract call txs in this loop. Need to see if any exist by this point, since we may have run a valset update by now